threaded_examples.py

```diff
@@ -0,0 +1,103 @@
+"""Context propagation examples when dealing with multi-threaded
+applications.
+"""
+import threading
+
+from proposal import tracer, helpers
+from proposal.active_span_source import ThreadActiveSpanSource
+
+
+# use a specific ActiveSpanSource implementation
+tracer._active_span_source = ThreadActiveSpanSource()
+
+
+def main_thread_instrumented_only():
+    """The main thread is instrumented but not its children."""
+    def do_some_work():
+        # code executed in children threads; they don't have an ActiveSpan
+        assert tracer.active_span_source.active_span() is None
+
+    # code executed in the main thread
+    with tracer.start_active(operation_name='main_thread') as span:
+        threads = [threading.Thread(target=do_some_work) for _ in range(5)]
+        for t in threads:
+            t.start()
+
+        # ...do more work in the main thread...
+        for t in threads:
+            t.join(timeout=1)
+
+
+def main_thread_instrumented_children_continue():
+    """The main thread is instrumented and when it spawns
+    new threads, they continue the main thread trace.
+    """
+    def do_some_work():
+        # code executed in children threads; it must continue
+        # the trace started in another thread
+        assert tracer.active_span_source.active_span() is not None
+        with tracer.start_active(operation_name='child_thread') as span:
+            pass
+
+    # code executed in the main thread
+    with tracer.start_active(operation_name='main_thread') as span:
+        threads = [helpers.TracedThread(target=do_some_work) for _ in range(5)]
```

Comment:
If we don't use a "helper" or something (that may be part of the API) to explicitly continue the tracing across threads, here we'll end having 6 different not connected traces. This is true if the `ThreadActiveSpanSource` concrete implementation uses a thread local storage to keep the current ActiveSpan.